R/R/diagnostics.R

```diff
@@ -500,6 +501,24 @@ mdape <- function(df, w) {
 }
 
 
+#' Symmetric mean absolute percentage error
+#' based on Chen and Yang (2004) formula
+#'
+#' @param df Cross-validation results dataframe.
+#' @param w Aggregation window size.
+#'
+#' @return Array of symmetric mean absolute percent errors.
+#'
+#' @keywords internal
+smape <- function(df, w) {
+  sape <- abs(df$y - df$yhat) / ((abs(df$y) + abs(df$yhat)) / 2)
+  if (w < 0) {
+    return(data.frame(horizon = df$horizon, smape = sape))
+  }
+  return(rolling_median_by_h(x = sape, h = df$horizon, w = w, name = 'smape'))
```

Comment:
The SMAPE should be rolling_mean_by_h and not rolling_median_by_h, right?

Reply:
Definitely, sorry for that.